### PR TITLE
Logging changes for Loop.cpp

### PR DESCRIPTION
**Pull request by vatral**
As requested, the changes have been separated out.

### CODE REVIEW

File: nxcomp/Loop.cpp [hunk 1]
```diff
@@ -671,15 +671,11 @@ static int ReopenLogFile(char *name, ostream *&stream, int limit);
 static void handleCheckSessionInLoop();
 static void handleCheckBitrateInLoop();
 
-#if defined(TEST) || defined(INFO)
 static void handleCheckSelectInLoop(int &setFDs, fd_set &readSet,
-                                        fd_set &writeSet, T_timestamp selectTs);
-static void handleCheckResultInLoop(int &resultFDs, int &errorFDs, int &setFDs, fd_set &readSet,
+                                        fd_set &writeSet, T_timestamp selectTs);static void handleCheckResultInLoop(int &resultFDs, int &errorFDs, int &setFDs, fd_set &readSet,
```
Comment (sunweaver) [inline nxcomp/Loop.cpp]: missing line break here!

File: nxcomp/Loop.cpp [hunk 2]
```diff
@@ -671,15 +671,11 @@ static int ReopenLogFile(char *name, ostream *&stream, int limit);
...
                                         fd_set &writeSet, struct timeval &selectTs,
                                             struct timeval &startTs);
-static void handleCheckStateInLoop(int &setFDs);
-#endif
-
+	static void handleCheckStateInLoop(int &setFDs);
```
Comment (sunweaver) [inline nxcomp/Loop.cpp]: don't indent here, make commits as clean as possible, esp. such big ones...

File: nxcomp/Loop.cpp [hunk 3]
```diff
@@ -671,15 +671,11 @@ static int ReopenLogFile(char *name, ostream *&stream, int limit);
 static void handleCheckSessionInLoop();
 static void handleCheckBitrateInLoop();
 
-#if defined(TEST) || defined(INFO)
```
Comment (sunweaver) [inline nxcomp/Loop.cpp]: why do you drop ifdef TEST/INFO here?

**Review by sunweaver — CHANGES_REQUESTED** [review]
Please revisit the code passages I commented on. I did not really look deeply at the actual logging implementation, because that has been done by @Ionic on PR #193.

File: nxcomp/Loop.cpp (hunk 3 above)
Comment (Ionic) [reply nxcomp/Loop.cpp]: Because the logging code is not guarded by these tests anymore, so the functions must always be defined to make the code compile.

File: nxcomp/Loop.cpp [hunk 4]
```diff
@@ -14943,27 +13944,20 @@ static void handleCheckBitrateInLoop()
...
 
 static void handleCheckStateInLoop(int &setFDs)
-{
-  int fdLength;
-  int fdPending;
-  int fdSplits;
-
+{  
+  int fdLength;  int fdPending;  int fdSplits;
```
Comment (Ionic) [inline nxcomp/Loop.cpp]: Is that change intentional?

File: nxcomp/Loop.cpp [hunk 5]
```diff
@@ -2669,18 +2538,15 @@ int NXTransAlert(int code, int local)
...
     return 1;
   }
-  #if defined(DUMP) || defined(INFO)
   else
   {
     if (logofs == NULL)
     {
-      logofs = &cerr;
-    }
+      logofs = &cerr;    }
```
Comment (Ionic) [inline nxcomp/Loop.cpp]: Manual cleanup here.

File: nxcomp/Loop.cpp [hunk 6]
```diff
@@ -3389,33 +3189,27 @@ int SetupProxyConnection()
...
     listenSocket.getSpec(&socketUri);
-    *logofs << "Loop: listenSocket is "<< ( listenSocket.enabled() ? "enabled" : "disabled") << ". "
-            << "The socket URI is '"<< ( socketUri != NULL ? socketUri : "<unset>") << "'.\n" << logofs_flush;
-    free(socketUri);
-    socketUri = NULL;
-    #endif
+    nxinfo << "Loop: listenSocket is "<< ( listenSocket.enabled() ? "enabled" : "disabled") << ". "
+           << "The socket URI is '"<< ( socketUri != NULL ? socketUri : "<unset>") << "'.\n" << std::flush;
 
+    free(socketUri);    socketUri = NULL;
```
Comment (Ionic) [inline nxcomp/Loop.cpp]: Manual cleanup here.

File: nxcomp/Loop.cpp [hunk 7]
```diff
@@ -9225,36 +8536,30 @@ int ParseCommandLineOptions(int argc, const char **argv)
...
-  #ifdef TEST
 
-  *logofs << "Loop: Argc is " << argc << ".\n" << logofs_flush;
+  nxinfo << "Loop: Argc is " << argc << ".\n" << std::flush;
 
   for (int argi = 0; argi < argc; argi++)
-  {
-    *logofs << "Loop: Argv[" << argi << "] is " << argv[argi]
-            << ".\n" << logofs_flush;
+  {
```
Comment (Ionic) [inline nxcomp/Loop.cpp]: Stray whitespace after curly brace.

File: nxcomp/Loop.cpp [hunk 8]
```diff
@@ -13954,19 +13021,15 @@ void PrintProcessInfo()
...
   {
-    cerr << "Info" << ": Using errors file '" << errorsFileName << "'.\n";
-  }
+    cerr << "Info" << ": Using errors file '" << errorsFileName << "'.\n";  }
 
   if (*statsFileName != '\0')
   {
-    cerr << "Info" << ": Using stats file '" << statsFileName << "'.\n";
-  }
+    cerr << "Info" << ": Using stats file '" << statsFileName << "'.\n";  }
```
Comment (Ionic) [inline nxcomp/Loop.cpp]: The last two changes only remove correct newlines.

Because it's outputting directly to `cerr`, we should keep these lines in mind though. More on that later.

File: nxcomp/Loop.cpp [hunk 9]
```diff
@@ -14394,25 +13443,23 @@ int ParseArg(const char *type, const char *name, const char *value)
...
+         << "' from string '" << string << "' with base set to ";
 
   switch (tolower(*id))
   {
     case 'k':
     case 'm':
     case 'g':
-    {
-      *logofs << (char) toupper(*id);
+    {
```
Comment (Ionic) [inline nxcomp/Loop.cpp]: Stray whitespace after curly brace.

**Comment by Ionic** [comment]
While looking through your changes, I noticed that some logging output was duplicated by NoMachine - once outputted via their `logofs` infrastructure (which will be replaced by your new work) and once via `cerr`.

Do we want to keep this duplication or remove it? Also, what about logging output that only ever goes to `cerr` but was never written via `logofs`? Do we want to keep it as-is, or make it use your new logging capabilities?

**Comment by vatral** [comment]
@Ionic I've looked into the cerr/logofs bits. Many can be removed, and I've written some code to try to detect the duplicates, but in the end didn't enable it for this iteration because of the amount of changes that need careful checking.

My plan was to follow up with removing the duplication.

**Review by Ionic — APPROVED** [review]

**Comment by Ionic** [comment]
Rebased against current `improved-logging-loop` branch.

I also fixed all whitespace errors that I could find.

**Review by sunweaver — APPROVED** [review]
LGTM
